### PR TITLE
add documentation for tidb_opt_distinct_agg_push_down

### PR DESCRIPTION
**Pull request by SunRunAway**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):https://github.com/pingcap/tidb/issues/14623

### CODE REVIEW

File: reference/configuration/tidb-server/tidb-specific-variables.md [hunk 1]
```diff
@@ -43,6 +43,13 @@ set @@global.tidb_distsql_scan_concurrency = 10
 - This variable is used to set whether the optimizer executes the optimization operation of pushing down the aggregate function to the position before Join.
 - When the aggregate operation is slow in query, you can set the variable value to 1.
 
+### tidb_opt_distinct_agg_push_down
+
+- Scope: SESSION
+- Default value: 0
+- This variable is used to set whether the optimizer executes the optimization operation of pushing down the aggregate function with distinct(like `select count(distinct a) from t`) to Coprocessor.
+- When the aggregate with distinct operation is slow in query, you can set the variable value to 1.
```
Comment (zz-jason) [inline reference/configuration/tidb-server/tidb-specific-variables.md:51]: please add an example to describe how to confirm that the `distinct` is pushed down.

**Comment by TomShawn** [comment]
@zz-jason PTAL again
@SunRunAway Please fix the CI failure.

**Comment by SunRunAway** [comment]
> @zz-jason PTAL again
> @SunRunAway Please fix the CI failure.

Fixed.

File: reference/configuration/tidb-server/tidb-specific-variables.md [hunk 2]
```diff
@@ -43,6 +43,41 @@ set @@global.tidb_distsql_scan_concurrency = 10
 - This variable is used to set whether the optimizer executes the optimization operation of pushing down the aggregate function to the position before Join.
 - When the aggregate operation is slow in query, you can set the variable value to 1.
 
+### tidb_opt_distinct_agg_push_down
+
+- Scope: SESSION
+- Default value: 0
+- This variable is used to set whether the optimizer executes the optimization operation of pushing down the aggregate function with distinct(like `select count(distinct a) from t`) to Coprocessor.
```
Comment (TomShawn) [inline reference/configuration/tidb-server/tidb-specific-variables.md:50]:
```suggestion
- This variable is used to set whether the optimizer executes the optimization operation of pushing down the aggregate function with `distinct` (such as `select count(distinct a) from t`) to Coprocessor.
```

File: reference/configuration/tidb-server/tidb-specific-variables.md [hunk 3]
```diff
@@ -43,6 +43,41 @@ set @@global.tidb_distsql_scan_concurrency = 10
...
+- When the aggregate with distinct operation is slow in query, you can set the variable value to 1.
```
Comment (TomShawn) [inline reference/configuration/tidb-server/tidb-specific-variables.md:51]:
```suggestion
- When the aggregate function with the `distinct` operation is slow in the query, you can set the variable value to `1`.
```

File: reference/configuration/tidb-server/tidb-specific-variables.md [hunk 4]
```diff
@@ -43,6 +43,41 @@ set @@global.tidb_distsql_scan_concurrency = 10
...
+
+In the following example, before `tidb_opt_distinct_agg_push_down` is on, tidb needs to read all data from tikv and execute `disctinct` on tidb side. After `tidb_opt_distinct_agg_push_down` is turned on,` distinct a` is pushed down to Coprocessor, and a group by column `test.t.a` is added to `HashAgg_5`.
```
Comment (TomShawn) [inline reference/configuration/tidb-server/tidb-specific-variables.md:53]:
```suggestion
In the following example, before `tidb_opt_distinct_agg_push_down` is enabled, TiDB needs to read all data from TiKV and execute `disctinct` on the TiDB side. After `tidb_opt_distinct_agg_push_down` is enabled, `distinct a` is pushed down to Coprocessor, and a `group by` column `test.t.a` is added to `HashAgg_5`.
```

File: reference/configuration/tidb-server/tidb-specific-variables.md [hunk 5]
```diff
@@ -43,6 +43,41 @@ set @@global.tidb_distsql_scan_concurrency = 10
...
+
+```
```
Comment (TomShawn) [inline reference/configuration/tidb-server/tidb-specific-variables.md:55]:
```suggestion
```sql
```

**Comment by TomShawn** [comment]
LGTM
@zz-jason PTAL

**Pull request merged**